exporter/metrics_exporter.go

```diff
@@ -141,19 +147,35 @@ func (m *metricsExporter) recordMetrics() {
 		// concurrently, so we assume that as long as that cap isn't enormous (which
 		// would only occur on an enormous cluster), it's practical to iterate over
 		// all the running workers.
+		var countError error
 		if phase == core.WorkerPhaseRunning {
 			var pendingJobs int
-			// TODO: This logic doesn't currently account for the possibility that
-			// there are more pages of events with running workers, so this count
-			// of pending jobs could be wrong.
-			for _, event := range events.Items {
-				for _, job := range event.Worker.Jobs {
-					if job.Status.Phase == core.JobPhasePending {
-						pendingJobs++
+			for {
+				for _, event := range events.Items {
+					for _, job := range event.Worker.Jobs {
+						if job.Status.Phase == core.JobPhasePending {
+							pendingJobs++
+						}
 					}
 				}
+				if events.Continue == "" {
+					break
+				}
+				events, countError = m.apiClient.Core().Events().List(
+					context.Background(),
+					&core.EventsSelector{
+						WorkerPhases: []core.WorkerPhase{phase},
+					},
+					&meta.ListOptions{Continue: events.Continue},
+				)
+				if countError != nil {
+					log.Println(countError)
+					break
+				}
```

Comment:
Error handling is more clearly coupled to the `m.apiClient.Core().Events().List(...)` function call when written like this:

```
if events, countError = m.apiClient.Core().Events().List(
  context.Background(),
  &core.EventsSelector{
    WorkerPhases: []core.WorkerPhase{phase},
  },
  &meta.ListOptions{Continue: events.Continue},
); countError != nil {
  log.Println(countError)
  break
}
```